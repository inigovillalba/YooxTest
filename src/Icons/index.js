export { default as Audio } from "./Audio/Audio";
export { default as Delivery } from "./Delivery/Delivery";
export { default as Edit } from "./Edit/Edit";
export { default as Favourites } from "./Favourites/Favourites";
export { default as GiftCards } from "./GiftCards/GiftCards";
