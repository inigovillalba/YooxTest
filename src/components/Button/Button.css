.btn-container {
  font-family: Arial, Helvetica, sans-serif;
}

.btn-container.medium {
  min-width: 190px;
}

.btn-container.large {
  min-width: 256px;
}

.btn-container .btn {
  width: 100%;
  height: 40px;
  background-color: #000000;
  border: 0;
  color: #ffffff;
  border-radius: 2px;
  font-size: 1em;
}

.btn-container .btn:focus {
  outline: 0;
}

.btn-container .btn:hover {
  background-color: #4f4f4f;
  cursor: pointer;
}
