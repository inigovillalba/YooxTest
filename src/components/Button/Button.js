import React, { Component } from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

import Tooltip from "../Tooltip/Tooltip";

import "./Button.css";

class Button extends Component {
  constructor(props) {
    super(props);
    this.state = {
      tooltipActive: false
    };
  }

  static propTypes = {
    size: PropTypes.string,
    buttonText: PropTypes.string.isRequired,
    tooltipItems: PropTypes.arrayOf(PropTypes.object)
  };

  static defaultProps = { size: "medium", tooltipItems: [] };

  componentDidMount() {
    window.addEventListener("click", this.handleClick);
  }

  handleClick = e => {
    const { type } = e.target;

    if (type !== "button") {
      this.setState({ tooltipActive: false });
    }
  };

  toggleTooltip = () => {
    const { tooltipItems } = this.props;

    if (!tooltipItems.length) {
      return;
    }
    this.setState({ tooltipActive: !this.state.tooltipActive });
  };

  render() {
    const { tooltipActive } = this.state;
    const { buttonText, tooltipItems, size } = this.props;

    const btnContainerClass = classNames({
      "btn-container": true,
      small: size === "small",
      medium: size === "medium",
      large: size === "large"
    });

    return (
      <div className={btnContainerClass}>
        <button className="btn" onClick={this.toggleTooltip} type="button">
          {buttonText}
        </button>
        {tooltipItems && (
          <Tooltip active={tooltipActive} items={tooltipItems} />
        )}
      </div>
    );
  }
}

export default Button;
