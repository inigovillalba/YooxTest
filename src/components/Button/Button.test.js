/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import Button from "./Button";
import { Audio, Delivery, Edit, Favourites, GiftCards } from "../../Icons";

describe("Button", () => {
  test("renders correctly", () => {
    const renderer = new ShallowRenderer();
    const tooltipData = {
      buttonText: "Button CTA",
      size: "large",
      tooltipItems: [
        { id: 0, icon: Audio, text: "Button Call To Action Longer" },
        { id: 1, icon: Delivery, text: "Button Call To Action Longer" },
        { id: 2, icon: Edit, text: "Button Call To Action Longer" },
        { id: 3, icon: Favourites, text: "Button Call To Action Longer" },
        { id: 4, icon: GiftCards, text: "Button Call To Action Longer" }
      ]
    };
    const wrapper = renderer.render(<Button {...tooltipData} />);

    expect(wrapper).toMatchSnapshot();
  });
});
