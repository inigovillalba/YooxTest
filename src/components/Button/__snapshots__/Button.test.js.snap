// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div
  className="btn-container large"
>
  <button
    className="btn"
    onClick={[Function]}
    type="button"
  >
    Button CTA
  </button>
  <Tooltip
    active={false}
    items={
      Array [
        Object {
          "icon": [Function],
          "id": 0,
          "text": "Button Call To Action Longer",
        },
        Object {
          "icon": [Function],
          "id": 1,
          "text": "Button Call To Action Longer",
        },
        Object {
          "icon": [Function],
          "id": 2,
          "text": "Button Call To Action Longer",
        },
        Object {
          "icon": [Function],
          "id": 3,
          "text": "Button Call To Action Longer",
        },
        Object {
          "icon": [Function],
          "id": 4,
          "text": "Button Call To Action Longer",
        },
      ]
    }
  />
</div>
`;
